File: server/datasets/python_source-code/Multiple-B2/organic_8.py
# Two Sum (hash map) - organic submission 8
# Compact style using the walrus operator with dict.get().

def two_sum(nums, target):
    seen = {}
    for i, num in enumerate(nums):
        if (match := seen.get(target - num)) is not None:
            return [match, i]
        seen[num] = i
    return None


if __name__ == "__main__":
    nums = [1, 2, 3, 9]
    target = 11
    print("Indices:", two_sum(nums, target))

File: server/datasets/java_source-code/Multiple-B2/organic_8.java
// Two Sum (hash map) - organic submission 8
// Clean compact style using getOrDefault / get.

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            Integer match = seen.get(target - nums[i]);
            if (match != null) {
                return new int[]{match, i};
            }
            seen.put(nums[i], i);
        }
        return null;
    }

    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 9};
        int target = 11;
        System.out.println("Indices: " + Arrays.toString(twoSum(nums, target)));
    }
}
